### ISSUE TITLE
Migrating non kSecClassGenericPassword item results in data loss

### ISSUE THREAD

**Issue by MarkVillacampa** [square/Valet#137]
Since all items stored by Valet are of type `kSecClassGenericPassword`, wouldn't migrating an item of a different type like `kSecClassInternetPassword` loose attribute information like `kSecAttrServer`?

**Comment by dfed** [comment]
Good point! Maybe we could add a requirement that `query` provided to `migrateObjects` had the `kSecClass as String : kSecClassGenericPassword as String` as a key:value pair. Seems like a good safety net.

In the meantime though, sending a query to migrate values that are not `kSecClassGenericPassword` is considered user error. I'm going to adjust the title of this issue accordingly, and label this an enhancement request.

If you'd like to submit a PR to `ErrorHandler.assert` that the `migrateObjects` `query` has `kSecClassGenericPassword`, we'd accept it. Just make sure to add a unit test or two :)
